File: Practice.Sharp/Practice.Sharp.Fsharp/Validators.fs
module Validators
open Monads
open System
open Constants

let evaluateType (number: string) = match Int32.TryParse(number) with
                                        | true, num -> Success (num)
                                        | false, _ -> Failure InvalidIntType

let evaluateRange   (min: int) 
                    (max: int) 
                    (number: int) = match number >= min && number <= max with
                                        |true  -> Success number
                                        |false -> Failure InvalidRange

let evaluateArrayLength (min: int)
                        (max: int)
                        (arr: int[])  =
    match arr.Length >= min && arr.Length <= max with
    | true -> Success arr
    | false -> Failure InvalidArrayRange

let evaluateArrayMax    (max: int) 
                        (arr: int[]) =
    Array.max arr
        |> function
            | n when n <= max -> Success arr
            | _ -> Failure InvalidArrayMax

let evaluateArrayMin    (min: int) 
                        (arr: int[]) =
  Array.min arr
        |> function
            | n when n >= min -> Success arr
            | _ -> Failure InvalidArrayMin

File: Practice.Sharp/Practice.Sharp.Csharp/Validators.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Practice.Sharp.Csharp
{
    public static class Validators
    {
        public static bool IsInRange(this int num, int min, int max)
        {
            return num >= min && num <= max;
        }

        public static bool IsInRange(this int[] arr, int min, int max)
        {
            return arr.Length >= min && arr.Length <= max;
        }
    }
}
